omegaconf/dictconfig.py

```diff
@@ -682,3 +683,49 @@ def _dict_conf_eq(d1: "DictConfig", d2: "DictConfig") -> bool:
                 return False
 
         return True
+
+    def _to_object(self) -> Any:
+        """Instantiate an instance of `self._metadata.object_type`.
+        This requires `self` to be a structured config.
+        Nested subconfigs are converted to_container with resolve=True."""
+        from ._utils import get_structured_config_field_names
+
+        object_type = self._metadata.object_type
+        assert is_structured_config(object_type)
+        object_type_field_names = set(get_structured_config_field_names(object_type))
+
+        field_items: Dict[str, Any] = {}
+        nonfield_items: Dict[str, Any] = {}
+        for k in self.keys():
+            node = self._get_node(k)
+            assert isinstance(node, Node)
+            node = node._dereference_node(throw_on_resolution_failure=True)
+            assert node is not None
```

Comment:
@omry calling `get_node()` followed by `_dereference_node()` is a somewhat common operation and it's a bit annoying right now with the mypy asserts.
Would it make sense to add a `get_dereferenced_node()` function that would return a `Node` (not optional) to streamline such code?

Reply:
That would make sense to me.
I'll do it in another PR :)

Reply:
First step is here: #668